### PR TITLE
Obj_Hamishi OK

### PR DESCRIPTION
**Pull request by Zelllll**
This is the bronze boulder that can only be broken with the Megaton Hammer.

### CODE REVIEW

File: src/overlays/actors/ovl_Obj_Hamishi/z_obj_hamishi.c [hunk 1]
```diff
@@ -30,17 +29,175 @@ const ActorInit Obj_Hamishi_InitVars = {
...
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+
+    if (globalCtx->csCtx.state != 0) {
+        this->actor.uncullZoneForward += 1000.0f;
+    }
+    if (this->actor.shape.rot.y == 0) {
+        newRotY = Math_Rand_ZeroFloat(65536.0f);
+        this->actor.initPosRot.rot.y = newRotY;
+        this->actor.posRot.rot.y = newRotY;
+        this->actor.shape.rot.y = newRotY;
```
Comment (Roman971) [inline src/overlays/actors/ovl_Obj_Hamishi/z_obj_hamishi.c:142]: You can probably avoid this temp bydoing something like this:
```suggestion
        this->actor.shape.rot.y =  this->actor.posRot.rot.y = this->actor.initPosRot.rot.y = Math_Rand_ZeroFloat(65536.0f);
```

File: src/overlays/actors/ovl_Obj_Hamishi/z_obj_hamishi.c [hunk 2]
```diff
@@ -30,17 +29,175 @@ const ActorInit Obj_Hamishi_InitVars = {
...
+    }
+
+    ObjHamishi_InitCollision(&this->actor, globalCtx);
+    func_80061ED4(&this->actor.colChkInfo, NULL, &sColChkInfoInit);
+
+    if (Flags_GetSwitch(globalCtx, this->actor.params & 0x3F) != 0) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_Obj_Hamishi/z_obj_hamishi.c:148]:
```suggestion
    if (Flags_GetSwitch(globalCtx, this->actor.params & 0x3F)) {
```

File: src/overlays/actors/ovl_Obj_Hamishi/z_obj_hamishi.c (hunk 1 above)
Comment (Zelllll) [reply src/overlays/actors/ovl_Obj_Hamishi/z_obj_hamishi.c:142]: Good idea, I'll test if that matches and then commit it if so.

**Comment by Zelllll** [comment]
Alright I made all the changes Roman suggested.

**Review by Roman971 — APPROVED** [review]

**Review by fig02 — APPROVED** [review]

**Pull request merged**